ext/crypto/00_crypto.js

```diff
@@ -2191,50 +2195,140 @@
     return key;
   }
 
-  function importKeyECDSA(
+  const SUPPORTED_EC_KEY_USAGES = {
+    "ECDSA": {
+      public: ["verify"],
+      private: ["sign"],
+      jwtUse: "sig",
+    },
+    "ECDH": {
+      public: [],
+      private: ["deriveKey", "deriveBits"],
+      jwtUse: "enc",
+    },
+  };
+
+  function importKeyEC(
     format,
     normalizedAlgorithm,
     keyData,
     extractable,
     keyUsages,
   ) {
+    const supportedUsages = SUPPORTED_EC_KEY_USAGES[normalizedAlgorithm.name];
+
+    // 1.
+    if (
+      !ArrayPrototypeIncludes(
+        supportedNamedCurves,
+        normalizedAlgorithm.namedCurve,
+      )
+    ) {
+      throw new DOMException(
+        "Invalid namedCurve",
+        "DataError",
+      );
+    }
```

Comment:
Ordering is important, because this is observable. This check happens as the first step for the "raw" algorithm, but only later for JWK.

Please move this to all the relevant spots in the different algorithms.